### ISSUE TITLE
extending fields in the Telemetry/Alert feed

### ISSUE THREAD

**Issue by nyrahul** [kubearmor/KubeArmor#532]
### Short Description

Currently, the process name is part of the telemetry/alert feeds. We need process full path as part of the same feed.

<details>
<summary>Sample Log feed</summary>

```
== Log / 2021-12-12 19:48:37.737160 ==
Cluster Name: Default
Host Name: pandora
Namespace Name: default
Pod Name: log4j-kubearmor
Container ID: 7ccca0b0a09ba86c96d581f695a534d0ec1d7a844f30efc16e0e72568a84cc39
Container Name: log4j-kubearmor
Type: ContainerLog
Source: jspawnhelper
Operation: Process
Resource: /bin/touch /tmp/log4jServerp0wn3d
Data: syscall=SYS_EXECVE
Result: Passed
```
Note the field `Source` ... This is the process name. We need process path as well.
</details>

<details>
<summary>Sample Alert feed</summary>

```
== Alert / 2021-12-12 19:57:07.871126 ==
Cluster Name: Default
Host Name: pandora
Namespace Name: default
Pod Name: log4j-kubearmor
Container ID: 7ccca0b0a09ba86c96d581f695a534d0ec1d7a844f30efc16e0e72568a84cc39
Container Name: log4j-kubearmor
Policy Name: do-not-allow-exec-from-java
Severity: 5
Message: disallowed execing from java process
Type: MatchedPolicy
Source: jspawnhelper
Operation: Process
Resource: /bin/touch /tmp/log4jServerp0wn3d
Data: syscall=SYS_EXECVE
Action: Block
Result: Passed
```
Note the field `Source` ... This is the process name. We need process path as well.
</details>

Tasks:

- [ ] get complete process path along with process name ... have these are two different fields
- [ ] For host-policies we have PPID and PID fields with process IDs .. apart from IDs we need corresponding process names/paths.
- [x] we receive following field when open is attempted, `syscall=SYS_OPENAT fd=100 flags=/path/to/file` ... the flags should be the flags as mentioned in [the man page](https://linux.die.net/man/2/openat).

### Is your feature request related to a problem? Describe the use case.

The Process Path would be needed if a policy discovery has to be made based on the parent process.

### Describe the solution you'd like

Add `ProcessPath` or `Source Path` as a field in the telemetry/alert feeds.

CC: @weirdwiz

**Comment by nyrahul** [comment]
Sample host policy alert:
``` json
{
  "Timestamp": 1639803960,
  "UpdatedTime": "2021-12-18T05:06:00.077564Z",
  "ClusterName": "Default",
  "HostName": "pandora",
  "HostPID": 3390423,
  "PPID": 168556,
  "PID": 3390423,
  "UID": 1000,
  "PolicyName": "hsp-kubearmor-dev-proc-path-block",
  "Severity": "1",
  "Type": "MatchedHostPolicy",
  "Source": "zsh",
  "Operation": "Process",
  "Resource": "/usr/bin/sleep",
  "Data": "syscall=SYS_EXECVE",
  "Action": "Block",
  "Result": "Permission denied"
}
```

**Comment by h3llix** [comment]
```== Alert / 2021-12-29 16:33:52.353004 ==
Cluster Name: default
Host Name: minikube
Namespace Name: multiubuntu
Pod Name: ubuntu-1-deployment-5d6b975744-mcvdg
Container ID: 71984adde96f74292d770ea47b953dc9c44ef49d229d619de73a6db57773e6fe
Container Name: ubuntu-1-container
Policy Name: ksp-group-1-proc-path-block
Severity: 5
Message: block the sleep command
Type: MatchedPolicy
Source: /bin/bash
Operation: Process
Resource: /bin/sleep 1
Data: syscall=SYS_EXECVE
Action: Block
Result: Permission denied
```
In my case field `Source` already shows the `SourcePath`. @nyrahul Can you elaborate more on this ?

**Comment by nyrahul** [comment]
> In my case field `Source` already shows the `SourcePath`. @nyrahul Can you elaborate more on this ?

I think the full path will show up in most cases i.e., if the process does not specify a process name then by default the process path will be used by OS kerne. However, there are few processes which allow you to change the process name. The examples I quoted are such processes. In my example, `zsh` seems to be altering the process name, thus you can try with `zsh`.

**Comment by nam-jaehyun** [comment]
This issue is related to #501.

**Comment by daemon1024** [comment]
Fixed in #632
